core/controllers/learner_dashboard.py

```diff
@@ -189,6 +200,18 @@ class LearnerDashboardFeedbackThreadHandler(base.BaseHandler):
     """Gets all the messages in a thread."""
 
     GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+    URL_PATH_ARGS_SCHEMAS = {
+        'thread_id': {
+            'schema': {
+                'type': 'basestring',
+                'validators': [{
+                    'id': 'is_regex_matched',
+                    'regex_pattern': r'(exploration|collection)\.\w+\.\w+'
```

Comment:
I'm not sure whether it's a good idea to validate the id's pattern in the param validator layer (It's somewhat fine on the model-validation layer), I see the following issues:

1. We do have some demo activities (exp/collection) with numeric ids on the server. Whether this handler work for it
2. Makes it cohesive with the internal structure of generating id and will make it hard to add a test for this handler.
3. We didn't have this regex check for it's in the ACL layer before, are we expecting this PR to "intentionally" add new validations for the handler?

Reply:
@DubeySandeep thank you for providing feedback, @vojtechjelinek and @rohitkatlaa can you please provide suggestion on this ?
Thank you.

Reply:
1. Demos should be fine, we only require at least one alphanumeric char in some parts of the ID, which even exp with ID '1', should conform.
2. I do't think it will be that hard, it seems to be passing tests now.
3. I think the purpose of this project is to add validation in places where they are missing or others can be added.